plugins/lsp/language-servers/efmls-configs.nix

```diff
@@ -161,71 +50,45 @@ in
         };
       }
     */
-    setup =
-      let
-        languageTools =
-          lang: kind: map (v: v.name) (if hasAttr kind tools.${lang} then tools.${lang}.${kind} else [ ]);
-
-        miscLinters = languageTools "misc" "linters";
-        miscFormatters = languageTools "misc" "formatters";
-
-        mkChooseOption =
-          lang: kind: possible:
-          let
-            toolType = with types; either (enum possible) helpers.nixvimTypes.rawLua;
-          in
-          mkOption {
-            type = with types; either toolType (listOf toolType);
-            default = [ ];
-            description = "${kind} tools for ${lang}";
-          };
-      in
-      mkOption {
-        type = types.submodule {
-          freeformType = types.attrs;
-
-          options =
-            (listToAttrs (
-              map (
-                lang:
-                let
-                  langTools = languageTools lang;
-                in
-                {
-                  name = lang;
-                  value = {
-                    linter = mkChooseOption lang "linter" ((langTools "linters") ++ miscLinters);
-                    formatter = mkChooseOption lang "formatter" ((langTools "formatters") ++ miscFormatters);
-                  };
-                }
-              ) languages
-            ))
-            // {
-              all = {
-                linter = mkChooseOption "all languages" "linter" miscLinters;
-                formatter = mkChooseOption "all languages" "formatter" miscFormatters;
-              };
-            };
-        };
-        description = "Configuration for each filetype. Use `all` to match any filetype.";
-        default = { };
+    setup = lib.mkOption {
```

Comment:
Is this a good opportunity to transition to using `settings` (and maybe mkPlugin or at least mkSettingsOption)?

If not, let's add a TODO comment

Reply:
Well, not really, because this plugin is not really configured with a setup option, it plugs directly into nvim-lspconfig & we do some mapping of the options.
So I'd say this is a plugin that does not fit at all in the RFC-42 schema

Reply:
I might be missing something, but I think doing `plugins.lsp.servers.efm.extraOptions.settings.languages = cfg.settings` would be fairly idiomatic?

We'd just have to make sure the settings option's description talked about that option instead of `require(..).setup`.